### ISSUE TITLE
[Bug]: Tooltip and DefinitionTooltips don't work well in tables with sticky headers

### ISSUE THREAD

**Issue by valente** [carbon-design-system/carbon#14428]
### Package

@carbon/react

### Browser

Chrome

### Package version

1.35.0

### React version

18.2.0

### Description

TooltipDefinition & Tooltips break tables with sticky elements, making the trigger elements fly over sticked headers.

### Reproduction/example

https://stackblitz.com/edit/github-737hdm-zg2ygf

### Steps to reproduce

horizontally scroll the table

### Suggested Severity

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by valente** [comment]
First, updates from my side: I found a workaround by adding a zindex declaration to the sticky column. Not sure why this is works but it seems to "revert" the effect of adding the tooltip.

About your response: this report was about a sticky column, not a sticky row. I think you referred to a sticky row/header.

Thank you for sharing the existence of a more comprehensive solution. I'm happy to see that, since I also continously face popover/tooltips issues in carbon-react.

In short, I consider this worked-around on my side, but I'm not sure this issue should be closed. Not sure adding a tooltip should have had this effect in the first place.

**Comment by valente** [comment]
Here's a fork with the fix https://stackblitz.com/edit/github-737hdm-mcucbi?file=src%2Findex.scss

**Comment by guidari** [comment]
Fixed in #14493
Here it is an working example in [Stackblitz](https://stackblitz.com/edit/github-yfpqcc?file=src%2FApp.jsx,src%2Findex.scss). The only thing to be aware of is the content inside the td with the `position: sticky`. The content overflows the `DefinitionTooltip`. With a few changes in the CSS you might fixed.
